universum/modules/artifact_collector.py

```diff
@@ -228,8 +278,30 @@ def collect_artifacts(self):
             self.structure.run_in_block(self.move_artifact, name, False, path, is_report=True)
         self.reporter.report_artifacts(list(self.collected_report_artifacts))
         for path in self.artifact_list:
-            name = "Collecting '" + os.path.basename(path) + "'"
-            self.structure.run_in_block(self.move_artifact, name, False, path)
+            if isinstance(path, ConditionalStepArtifacts):
+                self.collect_conditional_step_artifacts(path)
+            else:
+                name = "Collecting '" + os.path.basename(path) + "'"
+                self.structure.run_in_block(self.move_artifact, name, False, path)
+
+    def collect_conditional_step_artifacts(self, artifacts_info):
+        succeeded_artifact_path = artifacts_info.succeeded_config_artifact["path"]
+        succeeded_artifact_exists = os.path.exists(succeeded_artifact_path)
+        failed_artifact_path = artifacts_info.failed_config_artifact["path"]
+        failed_artifact_exists = os.path.exists(failed_artifact_path)
+
+        # At this point we don't know which branch step was executed,
+        # so assuming that any single branch artifact presence is good
+        if succeeded_artifact_exists and failed_artifact_exists:
+            raise CriticalCiException("Both conditional step branches artifacts exist")
```

Comment:
I think this logic is incorrect.
Generally we allow the same artifact paths to be specified multiple times in different steps. 
If some regular step has artifact path "a" (and creates corresponding artifact) and skipped conditional step has the same artifact path "a", this line would trigger the exception, which is incorrect.

Reply:
This code became redundant after the artifacts collection moment was changed from the end of the build to the end of each step.
Removed.